dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/JSONAPI.java

```diff
@@ -83,7 +88,69 @@ public ApiDocument build(UriInfo uriInfo) {
 			return doc;
 		}
 	}
-	
+
+	public static class ResourceObjectBuilderLinks<D> {
+		private ResponseData<D> data;
+		private List<Object> includes = Lists.newLinkedList();
+		private Map<String, Object> links = Maps.newHashMap();
+		private Object meta;
+
+		public ResourceObjectBuilderLinks(ResponseData<D> data) {
+			this.data = data;
+		}
+
+		public ResourceObjectBuilderLinks<D> include(Object include) {
+			includes.add(include);
+			return this;
+		}
+
+		public ResourceObjectBuilderLinks<D> link(String key, String link) {
+			links.put(key, link);
+			return this;
+		}
+
+		public ResourceObjectBuilderLinks<D> link(String key, JsonLink link) {
+			links.put(key, link);
+			return this;
+		}
+
+		public ResourceObjectBuilderLinks<D> meta(Object meta) {
+			this.meta = meta;
+			return this;
+		}
+
+		public ResourceObjectImplLinks<D> build() {
+			ResourceObjectImplLinks<D> doc = new ResourceObjectImplLinks<D>();
+			doc.setData(data);
+			doc.setType(getType(data.attributes));
+
+			if (!includes.isEmpty()) {
+				doc.setIncluded(includes);
+			}
+			if (!links.isEmpty()) {
+				doc.setLinks(links);
+			}
+			if (meta != null) {
+				doc.setMeta(meta);
+			}
+			return doc;
+		}
+
+		private String getType(D data) {
```

Comment:
don't like the name getType. Sounds like a typical getter... not too big a deal though.